packages/remix/src/client/useAwaitableNavigate.tsx

```diff
@@ -1,24 +1,22 @@
-import { useNavigate, useTransition } from '@remix-run/react';
+import { useLocation, useNavigate } from '@remix-run/react';
 import React from 'react';
 
 type Resolve = (value?: unknown) => void;
 
 export const useAwaitableNavigate = () => {
   const navigate = useNavigate();
-  const transition = useTransition();
+  const location = useLocation();
   const resolveFunctionsRef = React.useRef<Resolve[]>([]);
-
   const resolveAll = () => {
-    if (transition.state !== 'idle') {
-      return;
-    }
     resolveFunctionsRef.current.forEach(resolve => resolve());
     resolveFunctionsRef.current.splice(0, resolveFunctionsRef.current.length);
   };
 
+  // location.key will change even when navigating to the same url,
+  // so we will successfully resolve in that case as well
   React.useEffect(() => {
     resolveAll();
-  }, [transition.state]);
+  }, [location]);
```

Comment:
❓ Should this be `location.key` ?

Reply:
It could but it's used like that in react router docs so I think it's better this way. The key changing is just an observation.